server/log_rpc_server.go

```diff
@@ -541,6 +548,8 @@ func (t *TrillianLogRPCServer) GetLeavesByRange(ctx context.Context, req *trilli
 	}
 	defer t.closeAndLog(ctx, tree.TreeId, tx, "GetLeavesByRange")
 
+	t.fetchedLeaves.Add(float64(req.Count))
```

Comment:
The others are incremented after the leaves are fetched, but this one is before. It's also possible that this is incremented but we skip the fetch altogether because of the `req.StartIndex < int64(root.TreeSize)` check below.

I suggest always incrementing the counter immediately after fetching the leaves successfully, XOR incrementing it immediately before making the call regardless of whether it succeeds.

Reply:
Thanks, that was a mistake - there were all supposed to be after having fetched - but you've made me think that having it before making the actual call would give a better visibility of what we're asking storage to do.